### DISCUSSION TITLE
Feature Usage Limits and Quotas

### DISCUSSION THREAD

**Discussion by jbuijgers** [Unleash/unleash#3253] [Q&A] [answered]
Hey all,

I'm currently looking into multiple feature management solutions, after reading through the documentation, there's one thing I'm still struggling with: combing feature flags with usage limits[1] and quotas[2].

Am I right to be looking into feature management solutions to accomplish this? If so, would this be possible with Unleash?

1. Usage limits, an absolute number of usage, e.g. max. 10 users. If the customer created 10 users, disable the 'Add user' feature.
3. Quotas, an absolute number of usage per time-unit, e.g. max. 20 mails per month. If the customer has send 20 emails, disable the 'Send email' feature, and enable it at the end of the month.

**Comment by ivarconr** [comment, marked as answer]
Hi there! Good questions, and yes this can be accomplished using Unleash! 

You could use [custom context fields](https://docs.getunleash.io/reference/unleash-context#custom-context-fields) together with [strategy constraints](https://docs.getunleash.io/reference/strategy-constraints) to define such rules. 

1) Define a context field, call it "usersCount". Create a toggle and add a standard activation strategy with a numeric strategy constraint. It would look something like this:

![image](https://user-images.githubusercontent.com/158948/222950988-1ec06a54-6b0b-4c15-926c-0438e5cad3d5.png)

In you code, when using the SDK to evaluate the toggle, please remember to add "userCount" as part of the context when evaluating. 

2) Same as previous, but add a field for emails per month instead. 

Good luck!

**Reply by jbuijgers** [reply]
Thank you so much for the reply, this is perfect! :+1:
